agent/http.go

```diff
@@ -41,6 +42,18 @@ type HTTPServer struct {
 	proto string
 }
 
+type RedirectFS struct {
```

Comment:
Nitpick: public types and functions should really have a comment. We don't always do this in Consul currently but IMO it's a convention that's worth keeping. Example for this would be:

```go
// RedirectFS is an http.FileSystem implementation that ... We use it because...

// Open implements http.FileSystem
```